api/server/server.go

```diff
@@ -69,3 +74,14 @@ func initElasticsearch(config Config) {
 	}
 	log.Printf("connected to elasticsearch at %s\n", config.ElasticsearchURL)
 }
+
+// fail fast on initialization errors; there's no point in attempting
+// to continue in a degraded state if there are problems at start up
+func initKafka(config Config) {
+	log.Printf("connecting to kafka at %s\n", config.KafkaURL)
+	err := Kafka.Connect(config.KafkaURL)
+	if err != nil {
+		log.Fatalf("amplifer is unable to connect to kafka on: %s\n%v", config.KafkaURL, err)
```

Comment:
We generally should use `panic` / `log.Panicf` for these kinds of errors. I missed it on Chris' PR for initElasticSearch above, but I used `panic` for initEtcd. Panic will ensure any defer calls complete as it unwinds the stack. See: https://blog.golang.org/defer-panic-and-recover